src/predictor/gpu_predictor.cu

```diff
@@ -225,16 +225,39 @@ class GPUPredictor : public xgboost::Predictor {
     auto& offsets = *out_offsets;
     offsets.resize(devices_.Size() + 1);
     offsets[0] = 0;
+    auto host_data = data.ConstHostVector();
```

Comment:
I believe this copies the data, is this intended? Use of auto is tricky when dealing with references: https://stackoverflow.com/questions/7138588/c11-auto-what-if-it-gets-a-constant-reference

Reply:
No longer doing this as per @sriramch 's suggestion.